packages/migrations/src/clickhouse-actions/011-audit-logs.ts

```diff
@@ -0,0 +1,20 @@
+import type { Action } from '../clickhouse';
+
+export const action: Action = async exec => {
+  await exec(`
+    CREATE TABLE IF NOT EXISTS "audit_log"
+    (
+      "id" UUID DEFAULT generateUUIDv4() CODEC(ZSTD(1)),
+      "event_time" DateTime CODEC(ZSTD(1)),
+      "user_id" String CODEC(ZSTD(1)),
+      "user_email" String CODEC(ZSTD(1)),
+      "organization_id" String CODEC(ZSTD(1)),
+      "event_action" String CODEC(ZSTD(1)),
+      "metadata" String CODEC(ZSTD(1))
+    )
+    ENGINE = ReplacingMergeTree
+    ORDER BY ("organization_id")
+    TTL event_time + INTERVAL 1 YEAR
+    SETTINGS index_granularity = 8192
+  `);
```

Comment:
```suggestion
  await exec(`
    CREATE TABLE IF NOT EXISTS "audit_logs" (
      id String CODEC(ZSTD(1)),
      timestamp DateTime('UTC') CODEC(DoubleDelta, LZ4),
      organization_id LowCardinality(String) CODEC(ZSTD(1)),
      event_action LowCardinality(String) CODEC(ZSTD(1)),
      user_id String CODEC(ZSTD(1)),
      user_email String CODEC(ZSTD(1)),
      metadata String CODEC(ZSTD(1)),
      INDEX idx_action event_action TYPE set(0) GRANULARITY 1,
      INDEX idx_user_id user_id TYPE bloom_filter(0.001) GRANULARITY 1,
    )
    ENGINE = MergeTree
    PARTITION BY toYearWeek(timestamp, 1, 'UTC')
    ORDER BY (organization_id, timestamp)
    TTL timestamp + INTERVAL 1 YEAR
    SETTINGS index_granularity = 8192, ttl_only_drop_parts = 1;
  `);
  // Because we partition data by weeks,
  // I added `ttl_only_drop_parts` to simply drop parts,
  // instead of rows when TTL is exceeded.
```

There's quite a lot to unpack here.

The `ReplacingMergeTree` is not a good choice, especially with `ORDER BY ("organization_id")`. It would replace all rows of a given organization with the most recent row, we would lose data every time ClickHouse performs "merge of parts".

I also changed the `audit_log` to `audit_logs` plural - all the other tables are using this convention.

I replaced "event_time` with `timestamp` as we use this name for datetime columns everywhere, and I added a better codec.

The data partitioning was lacking, I added PARTITION BY `{year}-{week_number}`. If logs from 2024-01-20 are expired, ClickHouse will drop the full partition instead of each row individually. This is thanks to weekly-partiitioning and `ttl_only_drop_parts`.

The sort key is now `ORDER BY (organization_id, timestamp)`.
We always fetch audit logs for an organization, so it's the first "condition" in the sort key.
Then we order by timestamp, it's the second.
We could add a third one: event_action or user_id, but I see no perf benefits of it, to be honest.